aws/guardduty.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright © 2021 Cloud Posse, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	common "github.com/cloudposse/posse-cli/common/error"
+	"github.com/sirupsen/logrus"
+)
+
+func getGuardDutyClient(region string, role string) *guardduty.GuardDuty {
+	sess := GetSession()
+	creds := GetCreds(sess, role)
+	guardDutyClient := guardduty.New(sess, &aws.Config{Credentials: creds, Region: &region})
+
+	return guardDutyClient
+}
+
+func enableGuardDutyAdminAccount(client *guardduty.GuardDuty, accountID string) {
+	updateInput := guardduty.EnableOrganizationAdminAccountInput{AdminAccountId: &accountID}
+	client.EnableOrganizationAdminAccount(&updateInput)
+}
+
+// We need to enable GuardDuty in the AWS Organizations Management Account so that it can be added as a member
+// account in AWS GuardDuty's Administrator account. Accounts other than the Management Account don't need to be
+// excplicitly enabled, but the MA does.
+func enableGuardDutyInManagementAccount(client *guardduty.GuardDuty) {
+	_, err := client.CreateDetector(&guardduty.CreateDetectorInput{Enable: aws.Bool(true)})
+	if err != nil {
+		logrus.Error(err)
+	}
+}
+
+func isGuardDutyAdministratorAccountEnabled() bool {
```

Comment:
is this useful?

Reply:
d439a2d9e7f130d370c780e343dffaf43106cf99